tests/src/test/java/org/eclipse/hono/tests/registry/TenantManagementIT.java

```diff
@@ -158,6 +164,46 @@ public void testAddTenantSucceedsForEmptyBody(final VertxTestContext context) {
             .onComplete(context.completing());
     }
 
+    /**
+     * Verifies that the service successfully creates a tenant from a request having a tenant anchor
+     * with an ID and an another without an ID.
+     *
+     * @param context The Vert.x test context.
+     */
+    @Test
+    public void testAddTenantSucceedsForConfigurationWithMissingTrustAnchorIds(final VertxTestContext context) {
+        final PublicKey publicKey = TenantApiTests.getRandomPublicKey();
+        final TrustedCertificateAuthority trustAnchor1 = Tenants
+                .createTrustAnchor("test-ca", "CN=test-dn", publicKey.getEncoded(), publicKey.getAlgorithm(),
+                        Instant.now(), Instant.now().plus(365, ChronoUnit.DAYS));
+        final TrustedCertificateAuthority trustAnchor2 = Tenants
+                .createTrustAnchor(null, "CN=test-dn-1", publicKey.getEncoded(), publicKey.getAlgorithm(),
+                        Instant.now(), Instant.now().plus(365, ChronoUnit.DAYS));
+        final Tenant tenant = new Tenant()
+                .setTrustedCertificateAuthorities(List.of(trustAnchor1, trustAnchor2));
+
+        getHelper().registry.addTenant(
+                tenantId,
+                tenant,
+                "application/json",
+                HttpURLConnection.HTTP_CREATED)
+                .compose(ok -> getHelper().registry.getTenant(tenantId))
+                .onComplete(context.succeeding(httpResponse -> {
+                    context.verify(() -> {
+                        assertEquals(2,
+                                httpResponse.bodyAsJsonObject().getJsonArray(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA)
+                                        .size());
+                        assertEquals("test-ca", httpResponse.bodyAsJsonObject()
+                                .getJsonArray(RegistryManagementConstants.FIELD_PAYLOAD_TRUSTED_CA).getJsonObject(0)
+                                .getString(RegistryManagementConstants.FIELD_ID));
+                        assertNotNull(httpResponse.bodyAsJsonObject()
```

Comment:
IMHO we should use a local variable to store the value of `httpResponse.bodyAsJsonObject()` ...